pandas/tests/frame/methods/test_reindex.py

```diff
@@ -1258,3 +1258,28 @@ def test_invalid_method(self):
         msg = "Invalid fill method"
         with pytest.raises(ValueError, match=msg):
             df.reindex([1, 0, 2], method="asfreq")
+
+    def test_reindex_index_name_matches_multiindex_level(self):
+        df = DataFrame(
+            {"value": [1, 2], "other": ["A", "B"]},
+            index=Index([10, 20], name="a"),
+        )
+        target = MultiIndex.from_product(
+            [[10, 20], ["x", "y"]],
+            names=["a", "b"],
+        )
+
+        result = df.reindex(index=target)
+        expected = df.reindex(index=target, level="a")
```

Comment:
Could you build `expected` from a new DataFrame, e.g. `expected = DataFrame(data=..., index=..., ...)` instead of using `reindex`?